docs/DesignDocs/Attestation_API_Proposal.md

```diff
@@ -685,6 +685,109 @@ since legacy applications don't call these API functions explicitly.
+
+| Format ID | Evidence structure  |
+| -- | - |
+| 1. `OE_FORMAT_UUID_SGX_ECDSA_P256` | `oe_report_header || SGX_ECDSA_quote(hash) || custom_claims` |
+| 2. `OE_FORMAT_UUID_SGX_ECDSA_P256 + flag` | `oe_attestation_header || oe_report_header || SGX_ECDSA_quote(hash) || custom_claims` |
+| 3. `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT` | `oe_report_header || SGX_ECDSA_quote(custom_claims)` |
+| 4. `OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT + flag` | `oe_attestation_header || oe_report_header || SGX_ECDSA_quote(custom_claims)` |
+| 5. `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` | `SGX_ECDSA_quote(custom_claims)` |
+| 6. `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE + flag` | `oe_attestation_header || SGX_ECDSA_quote(custom_claims)` |
+
+For all evidence structures above, the hearts of the evidences are the same - SGX ECDSA_P256 quote. SGX ECDSA_P256 quote should to be like any other evidence format which is presented in 2 formats - with or without embedded format id. For example, format XYZ is presented as:
+1. Without flag - `Evidence_XYZ`
+2. With flag - `oe_attestation_header || Evidence_XYZ || custom_claims`
+
+That is, we only need to support these evidence structures:
+- Format 2: Generated by previous version oe_get_evidence().
+- Format 3: Generated by legacy oe_get_report() which will eventually be deprecated.
+- Format 5: Generated by SGX SDK.
+
+### User Experience
+
+#### Attester
+- To generate format 2:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    OE_EVIDENCE_FLAGS_EMBED_FORMAT_ID,
+    custom_claims,
+    size_t custom_claims_size,
+    ...);
+```
+
+- To generate format 3:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT,
+    0,  // Must be 0
+    custom_claims,
+    size_t custom_claims_size,  // Must be <= 64 bytes
+    ...);
+```
+
+- To generate format 5:
+```
+oe_get_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    0,
+    custom_claims, 
+    size_t custom_claims_size,  // Must be <= 64 bytes
+    ...);
+```
+
+#### Verifier
+- To verify format 2:
+```
+oe_verify_evidence(
+    NULL,
+    ...);
+```
+
+- To verify format 3:
+```
+oe_verify_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT,
+    ...);
+```
+
+- To verify format 5:
+```
+oe_verify_evidence(
+    OE_FORMAT_UUID_SGX_ECDSA_P256,
+    ...);
+```
+
+### Plan
+
+#### Phase 1 - V0.11
+- Deprecate `OE_FORMAT_UUID_SGX_ECDSA_P256_QUOTE` as proposed above.
+
+#### Phase 2 - V0.11
+- Remove `oe_report_header` from format 2. Increment `oe_attestation_header.version` to flag the change.
+- Modify `oe_verify_evidence()`, `oe_verify_report()`, and `oe_verify_attestation_certificate()` so they handle new format version.
+- Deprecate `oe_verify_attestation_certificate_with_evidence`.
+#### Phase 3 - V0.11
+- Deprecate `oe_get_report`.
+
+#### Phase 4 - Future
+- Deprecate OE_FORMAT_UUID_SGX_ECDSA_P256_REPORT. Define it as OE_FORMAT_UUID_SGX_ECDSA_P256.
+
 # Authors
 
 - Shanwei Cen (@shnwc)
+- Yen Lee (yenlee@microsoft.com)
```

Comment:
Seems inconsistent for one author to be listed by github id and another author to be listed by email address.  Can we be consistent, one way or the other?

Reply:
Agreed. My preference is github ID, which is public and will not change. Email addresses can be private and can change.

Reply:
Use github ID.